### PR TITLE
Add a cucumber feature to test link tag

### PR DESCRIPTION
**Pull request by ashmaroli**
Mainly to act as a control element for #6727 which passes other existing tests in spite of containing a breaking change.

### CODE REVIEW

File: features/step_definitions.rb [hunk 1]
```diff
@@ -36,7 +36,7 @@
 Given(%r!^I have an? "(.*)" page(?: with (.*) "(.*)")? that contains "(.*)"$!) do |file, key, value, text|
   File.write(file, Jekyll::Utils.strip_heredoc(<<-DATA))
     ---
-    #{key || "layout"}: #{value || "nil"}
+    #{key || "layout"}: #{value || "none"}
```
Comment (pathawks) [inline features/step_definitions.rb]: This is unrelated?

Comment (ashmaroli) [reply features/step_definitions.rb]: yep totally unrelated.. thought I'd fix this silently..

**Comment by DirtyF** [comment]
@jekyllbot: merge +dev
